src/ConfigReader/ConfigValuesVisitor.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace SprykerSdk\Integrator\ConfigReader;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\ArrayDimFetch;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Scalar;
+use PhpParser\Node\Stmt\Expression;
+use PhpParser\NodeVisitorAbstract;
+
+class ConfigValuesVisitor extends NodeVisitorAbstract
+{
+    /**
+     * @var string
+     */
+    protected const CONFIG_VAR_NAME = 'config';
+
+    /**
+     * @var array<string>
+     */
+    protected $configKeysToFind;
+
+    /**
+     * @var array<string, mixed>
+     */
+    protected $foundValues = [];
+
+    /**
+     * @param array $configKeysToFind
+     */
+    public function __construct(array $configKeysToFind)
+    {
+        $this->configKeysToFind = $configKeysToFind;
+    }
+
+    /**
+     * @param \PhpParser\Node $node
+     *
+     * @return \PhpParser\Node
+     */
+    public function enterNode(Node $node): Node
+    {
+        if (
+            $node instanceof Expression
+            && $node->expr instanceof Assign
+            && $node->expr->var instanceof ArrayDimFetch
+            && $node->expr->var->var instanceof Variable
+            && $node->expr->var->var->name === static::CONFIG_VAR_NAME
+            && $node->expr->var->dim instanceof ClassConstFetch
+            && $node->expr->var->dim->name instanceof Identifier
+            && in_array($this->getFullConstName($node->expr->var->dim), $this->configKeysToFind, true)
+        ) {
+            $this->foundValues[$this->getFullConstName($node->expr->var->dim)] = $this->getValuesFromAssignExpression($node->expr);
```

Comment:
have you tried applying the logic that checks values in the `$this->foundValues`. Maybe it could speed up even more our script?